### PR TITLE
Model/query for criteria and save

### PR DESCRIPTION
**Pull request by MelissaRoman06**

### CODE REVIEW

File: src/main/java/com/fundation/logic/common/JsonConverter.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ *
+ * This software is the confidential and proprietary information of Jalasoft.
+ * ("Confidential Information"). You shall not
+ * disclose such Confidential Information and shall use it only in
+ * accordance with the terms of the license agreement you entered into
+ * with Jalasoft.
+ */
+package com.fundation.logic.common;
+
+import com.fundation.logic.model.criteria.*;
```
Comment (titor-oopart) [inline src/main/java/com/fundation/logic/common/JsonConverter.java]: Put specific library

File: src/main/java/com/fundation/logic/model/CriteriaRecord.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    /**
+     * Allows to get the day the criteria was saved.
+     * @return Date the criteria was saved.
+     */
+    public String getDate() {
+        return this.date;
+    }
+
+    /**
+     * Allows to get criteria's type
```
Comment (titor-oopart) [inline src/main/java/com/fundation/logic/model/CriteriaRecord.java]: put dat at the end of the type

File: src/main/java/com/fundation/logic/model/criteria/Image.java [hunk 3]
```diff
@@ -12,10 +12,10 @@
 /**
  * Implements the model class Image and the getter and setter´s methods.
  *
- * @authors Andres Burgos.
- * @version 1.1
+ * @authors Andres Burgos
+ * @version 1.0
  */
-public class Image {
+public class Image extends Criteria{
```
Comment (titor-oopart) [inline src/main/java/com/fundation/logic/model/criteria/Image.java]: put space before {

File: src/main/java/com/fundation/logic/common/JsonConverter.java (hunk 1 above)
Comment (MelissaRoman06) [reply src/main/java/com/fundation/logic/common/JsonConverter.java]: done

File: src/main/java/com/fundation/logic/model/criteria/Image.java (hunk 3 above)
Comment (MelissaRoman06) [reply src/main/java/com/fundation/logic/model/criteria/Image.java]: done

File: src/main/java/com/fundation/logic/model/CriteriaRecord.java (hunk 2 above)
Comment (MelissaRoman06) [reply src/main/java/com/fundation/logic/model/CriteriaRecord.java]: done

File: src/main/java/com/fundation/logic/common/JsonConverter.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+                return gson.fromJson(json, Common.class);
+            case "Image":
+                return gson.fromJson(json, Image.class);
+            case "Video":
+                return gson.fromJson(json, Video.class);
+            default:
+                return null;
+        }
+    }
+}
```
Comment (titor-oopart) [inline src/main/java/com/fundation/logic/common/JsonConverter.java]: put an empty line

File: src/main/java/com/fundation/logic/model/QueryForCriteria.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        return criteriaRecords;
+    }
+
+    /**
+     * Prints all criteria in database.
+     */
+    public void showCriteriaInDB() {
+        List criteria = query.getAllCriterias();
+        for (int index = 0; index < criteria.size(); index++) {
+            System.out.println(criteria.get(index));
```
Comment (titor-oopart) [inline src/main/java/com/fundation/logic/model/QueryForCriteria.java]: delete this sout

File: src/main/java/com/fundation/logic/model/QueryForCriteria.java [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+     * Finds criteria that were saved between two given dates.
+     * @param date1 - Start date.
+     * @param date2 - End date.
+     */
+    public void findCriteria(Date date1, Date date2) {
+        String firstDate = date1.toString();
+        String secondDate = date2.toString();
+        List list = query.filterByDates(firstDate, secondDate);
+        for (int index = 0; index < list.size(); index++) {
+            System.out.println(list.get(index));
```
Comment (titor-oopart) [inline src/main/java/com/fundation/logic/model/QueryForCriteria.java]: delete this sout

File: src/main/java/com/fundation/logic/common/JsonConverter.java (hunk 4 above)
Comment (MelissaRoman06) [reply src/main/java/com/fundation/logic/common/JsonConverter.java]: done

File: src/main/java/com/fundation/logic/model/QueryForCriteria.java (hunk 5 above)
Comment (MelissaRoman06) [reply src/main/java/com/fundation/logic/model/QueryForCriteria.java]: done

File: src/main/java/com/fundation/logic/model/QueryForCriteria.java (hunk 6 above)
Comment (MelissaRoman06) [reply src/main/java/com/fundation/logic/model/QueryForCriteria.java]: done

**Pull request merged**